homeassistant/components/rituals_perfume_genie/sensor.py

```diff
@@ -0,0 +1,168 @@
+        """Initialize the fill sensor."""
+        super().__init__(diffuser, coordinator, FILL_SUFFIX)
+
+    @property
+    def icon(self):
+        """Return the fill sensor icon."""
+        if self.coordinator.data[HUB][SENSORS][FILL][ID] == FILL_NO_CARTRIDGE_ID:
+            return "mdi:beaker-question"
+        return "mdi:beaker"
+
+    @property
+    def state(self):
+        """Return the state of the fill sensor."""
+        return self.coordinator.data[HUB][SENSORS][FILL][TITLE]
+
+
+class DiffuserBatterySensor(DiffuserEntity):
+    """Representation of a diffuser battery sensor."""
+
+    def __init__(self, diffuser, coordinator):
+        """Initialize the battery sensor."""
+        super().__init__(diffuser, coordinator, BATTERY_SUFFIX)
+
+    @property
+    def state(self):
+        """Return the state of the battery sensor."""
+        # Use ICON because TITLE may change in the future.
+        # ICON filename does not match the image.
+        return {
+            "battery-charge.png": 100,
+            "battery-full.png": 100,
+            "battery-75.png": 50,
+            "battery-50.png": 25,
+            "battery-low.png": 10,
+        }[self.coordinator.data[HUB][SENSORS][BATTERY][ICON]]
+
+    @property
+    def _charging(self):
+        """Return battery charging state."""
+        return bool(
+            self.coordinator.data[HUB][SENSORS][BATTERY][ID] == BATTERY_CHARGING_ID
+        )
+
+    @property
+    def device_class(self):
+        """Return the class of the battery sensor."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def extra_state_attributes(self):
+        """Return the battery state attributes."""
+        return {
+            ATTR_BATTERY_LEVEL: self.coordinator.data[HUB][SENSORS][BATTERY][TITLE],
+            ATTR_BATTERY_CHARGING: self._charging,
+        }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the battery unit of measurement."""
+        return PERCENTAGE
+
+
+class DiffuserWifiSensor(DiffuserEntity):
+    """Representation of a diffuser wifi sensor."""
+
+    def __init__(self, diffuser, coordinator):
+        """Initialize the wifi sensor."""
+        super().__init__(diffuser, coordinator, WIFI_SUFFIX)
+
+    @property
+    def state(self):
+        """Return the state of the wifi sensor."""
+        # Use ICON because TITLE may change in the future.
+        return {
+            "icon-signal.png": 100,
+            "icon-signal-75.png": 70,
+            "icon-signal-low.png": 25,
+            "icon-signal-0.png": 0,
+        }[self.coordinator.data[HUB][SENSORS][WIFI][ICON]]
+
+    @property
+    def device_class(self):
+        """Return the class of the wifi sensor."""
+        return DEVICE_CLASS_SIGNAL_STRENGTH
+
+    @property
+    def extra_state_attributes(self):
+        """Return the wifi state attributes."""
+        return {
+            ATTR_SIGNAL_STRENGTH: self.coordinator.data[HUB][SENSORS][WIFI][TITLE],
```

Comment:
What measurement is this?

Reply:
Same as the battery. The description of the state as shown in the app: `low`, `medium`, ...

Reply:
Remove it completely or make it a separate sensor entity.